sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/storage/GcsResourceId.java

```diff
@@ -91,6 +91,17 @@ public String getScheme() {
     return GcsFileSystemRegistrar.GCS_SCHEME;
   }
 
+  @Override
+  public String getFilename() {
+    if (gcsPath.getNameCount() < 1) {
+      return null;
+    } else if (gcsPath.getNameCount() == 1) {
+      return gcsPath.getBucket();
+    } else {
+      return gcsPath.getFileName().toString();
+    }
```

Comment:
Yeah I was in two minds but included the bucket as it is part of the hierarchical component. I was going with anything after the the `scheme` should be included.